src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -0,0 +1,514 @@
+
+        public void Clear()
+        {
+            if (Count > 0)
+            {
+                Array.Clear(_buckets, 0, _buckets.Length);
+                Array.Clear(_entries, 0, Count);
+                Count = 0;
+            }
+            ++_version;
+        }
+
+        public bool ContainsKey(TKey key) => IndexOf(key) >= 0;
+
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity));
+            }
+
+            if (_entries.Length >= capacity)
+            {
+                return _entries.Length;
+            }
+            int newSize = HashHelpers.GetPrime(capacity);
+            Resize(newSize);
+            ++_version;
+            return newSize;
+        }
+
+        public TValue GetAt(int index) => GetAt(index, out _);
+
+        public TValue GetAt(int index, out TKey key)
+        {
+            if (index < 0 || index >= Count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index), "Index was out of range. Must be non-negative and less than the size of the collection.");
+            }
+
+            Entry entry = _entries[index];
+            key = entry.Key;
+            return entry.Value;
+        }
+
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        public TValue GetValue(TKey key)
+        {
+            int index = IndexOf(key);
+            if (index < 0)
+            {
+                throw new KeyNotFoundException("The given key was not present in the dictionary.");
+            }
+            return _entries[index].Value;
+        }
+
+        public int IndexOf(TKey key) => IndexOf(key, out _);
+
+        public void Insert(int index, TKey key, TValue value)
+        {
+            if (index < 0 || index > Count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index), "Index was out of range. Must be non-negative and less than or equal to the size of the collection.");
+            }
+
+            TryInsert(index, key, value, InsertionBehavior.ThrowOnExisting);
+        }
+
+        public bool Remove(TKey key) => Remove(key, out _);
+
+        public bool Remove(TKey key, out TValue value)
+        {
+            int index = IndexOf(key);
+            if (index >= 0)
+            {
+                value = _entries[index].Value;
+                RemoveAt(index);
+                return true;
+            }
+            value = default;
+            return false;
+        }
+
+        public void RemoveAt(int index)
+        {
+            int count = Count;
+            if (index < 0 || index >= count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index), "Index was out of range. Must be non-negative and less than the size of the collection.");
+            }
+
+            // Remove the entry from the bucket
+            UpdateBucket(index, true);
```

Comment:
Please use named parameter style as it saves looking at the callee. Eg `UpdateBucket(index, setToNext:true);`
Same elsewhere.
@stephentoub should that be in the [style guide](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md)?